flink-architecture-tests/flink-architecture-tests-test/src/main/java/org/apache/flink/architecture/rules/ITCaseRules.java

```diff
@@ -63,29 +66,41 @@
                                                     // after the JUnit 4->5 migration is closed.
-                                                    .or(
-                                                            containAnyFieldsInClassHierarchyThat(
-                                                                    arePublicStaticFinalOfTypeWithAnnotation(
-                                                                            MiniClusterWithClientResource
-                                                                                    .class,
-                                                                            ClassRule.class)))
-                                                    .or(
-                                                            containAnyFieldsInClassHierarchyThat(
-                                                                    arePublicFinalOfTypeWithAnnotation(
-                                                                            MiniClusterWithClientResource
-                                                                                    .class,
-                                                                            Rule.class))))));
+                                                    .or(miniClusterWithClientResourceClassRule())
+                                                    .or(miniClusterWithClientResourceRule()))));
+
+    private static DescribedPredicate<JavaClass> miniClusterWithClientResourceClassRule() {
+        return containAnyFieldsInClassHierarchyThat(
+                arePublicStaticFinalOfTypeWithAnnotation(
+                        MiniClusterWithClientResource.class, ClassRule.class));
+    }
+
+    private static DescribedPredicate<JavaClass> miniClusterWithClientResourceRule() {
+        return containAnyFieldsInClassHierarchyThat(
+                arePublicFinalOfTypeWithAnnotation(
+                        MiniClusterWithClientResource.class, Rule.class));
+    }
+
+    private static DescribedPredicate<JavaClass> miniClusterExtensionRule() {
+        return containAnyFieldsInClassHierarchyThat(
+                        arePublicStaticFinalOfType(MiniClusterExtension.class))
+                .and(
+                        containAnyFieldsInClassHierarchyThat(
+                                arePublicStaticFinalOfTypeWithAnnotation(
+                                        AllCallbackWrapper.class, RegisterExtension.class)));
+    }
+
+    private static DescribedPredicate<JavaClass> miniClusterWithClientExtensionRule() {
+        return containAnyFieldsInClassHierarchyThat(
+                        arePublicStaticFinalOfType(MiniClusterWithClientExtension.class))
+                .and(
+                        containAnyFieldsInClassHierarchyThat(
+                                arePublicStaticFinalOfTypeWithAnnotation(
+                                        AllCallbackWrapper.class, RegisterExtension.class)));
+    }
```

Comment:
how about with one method by using:
`containAnyFieldsInClassHierarchyThat(
                                                            arePublicStaticFinalOfType(
                                                                    MiniClusterExtension.class)).or(containAnyFieldsInClassHierarchyThat(
                                                            arePublicStaticFinalOfType(
                                                                    MiniClusterWithClientExtension.class))).and(
                        containAnyFieldsInClassHierarchyThat(
                                arePublicStaticFinalOfTypeWithAnnotation(
                                        AllCallbackWrapper.class, RegisterExtension.class)))`

Reply:
How important is that refactoring? It takes ages to regenerate the rule files.